### ISSUE TITLE
Training without ddp failed

### ISSUE THREAD

**Issue by shufujian** [CognitiveAISystems/MAPF-GPT#2]
Firstly, thank you for open sourcing such excellent research code.
But, i am facing some problem.
I train it without ddp, but the log tells me that `ddp_local_rank` is not defined.

``` shell
Traceback (most recent call last):
  File "train.py", line 377, in <module>
    X, Y = get_batch(train_data)  # fetch the very first batch
  File "train.py", line 251, in get_batch
    batch = next(data)
  File "train.py", line 199, in batch_generator
    start_index = ddp_local_rank * files_per_worker
NameError: name 'ddp_local_rank' is not defined
```

When i watch the source code again, i find that the [`ddp_local_rank`](https://github.com/CognitiveAISystems/MAPF-GPT/blob/main/train.py#L142) initialized only use ddp.
But in [batch_generate()](https://github.com/CognitiveAISystems/MAPF-GPT/blob/main/train.py#L191) the `ddp_local_rank`  is used when it is not initialized。

**Comment by Tviskaron** [comment]
Hi, shufujian! I'll investigate this and comeback with answer/fix.

**Comment by shufujian** [comment]
> Hi, shufujian! I'll investigate this and comeback with answer/fix.

Oh，thanks a lot！

And，I would also like to ask about the map generator mentioned in your article.

I'd be most grateful if you could answer me!

**Comment by Tviskaron** [comment]
The map generators are located at [link](https://github.com/CognitiveAISystems/pogema-toolbox/tree/main/pogema_toolbox/generators)

**Comment by shufujian** [comment]
> The map generators are located at [link](https://github.com/CognitiveAISystems/pogema-toolbox/tree/main/pogema_toolbox/generators)

Thank you very much for your help!

**Comment by Tviskaron** [comment]
I’ve just addressed the issue (see commit c44394660e8e1bdfbcbbf2f6042d0726a6ca18d1). Additionally, there’s now a much faster implementation of the dataset loader, which also consumes less GPU memory.

**Comment by shufujian** [comment]
> I’ve just addressed the issue (see commit [c443946](https://github.com/CognitiveAISystems/MAPF-GPT/commit/c44394660e8e1bdfbcbbf2f6042d0726a6ca18d1)). Additionally, there’s now a much faster implementation of the dataset loader, which also consumes less GPU memory.

Thank you very much!!!
